demos/sjtwo/eeprom/source/main.cpp

```diff
@@ -0,0 +1,50 @@
+#include <project_config.hpp>
+
+#include <cstdint>
+#include <iterator>
+
+#include "L1_Peripheral/lpc40xx/eeprom.hpp"
+#include "L2_HAL/displays/led/onboard_led.hpp"
+#include "utility/log.hpp"
+#include "utility/time.hpp"
+
+int main(void)
+{
+  LOG_INFO("Starting EEPROM Example");
+  constexpr size_t kPayloadSize = 128;
+  constexpr uint32_t kAddress   = 0b0110'0011'1000;
+
+  uint8_t * results = new uint8_t[kPayloadSize];
+
+  LOG_INFO("Initializing EEPROM");
+  sjsu::lpc40xx::Eeprom eeprom;
+  eeprom.Initialize();
+
+  // Generating Payload
+  uint8_t * list = new uint8_t[kPayloadSize];
+
+  for (size_t i = 0; i < kPayloadSize; i++)
+  {
+    list[i] = static_cast<uint8_t>(i*2);
+  }
+
+  while (true)
+  {
+    LOG_INFO("Starting Write");
+    eeprom.Write(list, kAddress, kPayloadSize);
+
+    LOG_INFO("Starting Read");
+    eeprom.Read(results, kAddress, kPayloadSize);
+
+    LOG_INFO("Results:");
+    for (size_t i = 0; i < kPayloadSize; i++)
+    {
+      LOG_INFO("%i: %u", i, results[i]);
+    }
+    LOG_INFO("");
+
+    sjsu::Delay(1000);
```

Comment:
Oh god! This will destroy the boards EEPROM if you leave this program running for a while. Maybe make this interactive so that the user has to press a button in order to write and read from the device. Basically pass the buck to the user of the board. Probably add a log message stating that the too many writes will cause the eeprom to fail.

Reply:
That's something I hadn't considered, but the button thing is a great idea. I currently don't have an SJ2 with buttons (only have the small one right now) so I can't test this yet. I'll try to get a hold of one and add this in.